internal/sdkv2provider/resource_cloudflare_workers_queue.go

```diff
@@ -0,0 +1,131 @@
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+)
+
+func resourceCloudflareWorkersQueue() *schema.Resource {
+	return &schema.Resource{
+		Schema:        resourceCloudflareWorkersQueueSchema(),
+		CreateContext: resourceCloudflareWorkersQueueCreate,
+		ReadContext:   resourceCloudflareWorkersQueueRead,
+		UpdateContext: resourceCloudflareWorkersQueueUpdate,
+		DeleteContext: resourceCloudflareWorkersQueueDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceCloudflareWorkersQueueImport,
+		},
+		Description: "Provides the ability to manage Cloudflare Workers Queue features.",
+	}
+}
+
+func resourceCloudflareWorkersQueueCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+
+	accountID := d.Get("account_id").(string)
+	if accountID == "" {
+		accountID = client.AccountID
+	}
+
+	queueName := d.Get("name").(string)
+	req := cloudflare.CreateQueueParams{
+		Name: queueName,
+	}
+
+	tflog.Debug(ctx, fmt.Sprintf("[Info] Creating Cloudflare Workers Queue from struct: %+v", req))
+
+	r, err := client.CreateQueue(ctx, cloudflare.AccountIdentifier(accountID), req)
+	if err != nil {
+		return diag.FromErr(errors.Wrap(err, "error creating workers queue"))
+	}
+
+	if r.ID == "" {
+		return diag.FromErr(fmt.Errorf("failed to find id in Create response; resource was empty"))
+	}
+
+	d.SetId(r.ID)
+
+	tflog.Info(ctx, fmt.Sprintf("Cloudflare Workers Queue ID: %s. Name: %s", d.Id(), queueName))
+
+	return resourceCloudflareWorkersQueueRead(ctx, d, meta)
+}
+
+func resourceCloudflareWorkersQueueRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	queueID := d.Id()
+
+	accountID := d.Get("account_id").(string)
+	if accountID == "" {
+		accountID = client.AccountID
+	}
+
+	resp, _, err := client.ListQueues(ctx, cloudflare.AccountIdentifier(accountID), cloudflare.ListQueuesParams{})
+	if err != nil {
+		return diag.FromErr(errors.Wrap(err, "error reading queues"))
+	}
+
+	var queue cloudflare.Queue
+	for _, r := range resp {
+		if r.ID == queueID {
+			queue = r
+			break
+		}
+	}
+
+	if queue.ID == "" {
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("account_id", accountID)
+
+	return nil
+}
+
+func resourceCloudflareWorkersQueueUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+
+	accountID := d.Get("account_id").(string)
+	if accountID == "" {
+		accountID = client.AccountID
+	}
+
+	// TODO(soon) fix the cloudflare-go UpdateQueue implementation: updating a queue should accept the existing name, as well as the new name. Other properties are read-only.
+	_, err := client.UpdateQueue(ctx, cloudflare.AccountIdentifier(accountID), cloudflare.UpdateQueueParams{
```

Comment:
this API feels kinda weird. if it's only using the `name` as the identifier, i would expect that changing it would be creating a new resource. in terraform, we can do this with `ForceNew: true` however, on a broader sense, any reason why we don't have a md5 style checksum for IDs?

Reply:
Queues are associated with a UUID, but the current V4 API endpoints all use the queue's name instead...

Currently, the "Update Queue" V4 endpoint (https://api.cloudflare.com/#queue-update-queue) is only capable of changing the queue's name. The existing name is in the URL (/queues/<existing-name>) while the desired new name is in the request body. The Queue's UUID remains unchanged though...

The cloudflare-go `UpdateQueue` function seems to be wrong currently as there is no way to pass the parameters correctly (existing name and queue name).